### DISCUSSION TITLE
DDT4All on macOS M1: Severe Lag When Editing ECUs After Scanning

### DISCUSSION THREAD

**Discussion by enesokyay** [cedricp/ddt4all#1205] [Q&A]
Hello,

I am using DDT4All on a MacBook with an M1 processor. The installation went smoothly, and the software runs without any initial issues. However, I am experiencing significant performance problems when attempting to edit any ECU after scanning.

Steps to Reproduce:

	1.	I connected my ELM327 OBD-II adapter (as shown in the attached image).
	2.	I launched DDT4All and successfully scanned all ECUs in my Renault Clio IV.
	3.	The scanning process works without any problems.
	4.	However, when I attempt to select and edit any ECU from the list, the application becomes extremely slow and unresponsive. The delay is so severe that I am unable to perform any meaningful operations.

Screenshots:

	•	I have attached screenshots of the application during the scanning process and the list of ECUs.
	•	Additionally, I have included a photo of the OBD-II adapter I am using.

System Information:

	•	MacBook Model: MacBook Pro with M1 Chip
	•	Operating System: macOS Ventura (or specify your version if different)
	•	DDT4All Version: 3.0.1 (Stable)

Troubleshooting Steps Already Taken:

	•	I have ensured that the software is up-to-date.
	•	I have tried restarting the application and the MacBook, but the issue persists.
	•	I have also tried using a different OBD-II adapter, but the problem remains the same.

Questions:

	•	Is this a known issue with the M1 processor, or could it be related to the compatibility of the software with macOS?
	•	Are there any suggested workarounds or settings that could improve performance?
	•	Would running DDT4All in a virtual machine (such as Parallels or VirtualBox) with Windows improve performance?

Thank you for your assistance!

![ IMG_1561](https://github.com/user-attachments/assets/13535e5a-8d1d-493d-be11-d671249b42b8) ![ IMG_97CD285E-496A-4F19-AD63-64EBB77E949A](https://github.com/user-attachments/assets/f4c6c302-4619-4368-8988-1df9c38880fc) ![ IMG_1559](https://github.com/user-attachments/assets/d5abdeff-b296-45d9-85a7-49d83010cc23)

**Comment by enesokyay** [comment]
I suspect that the issue might be related to the OBD adapter, so I have ordered a new one from AliExpress


6,64€ | ELM327 V2.1 OBD2 Tarayıcı TR6260S1 BT/Wifi ELM 327 Android /IOS PK Vgate Icar2 Kod Okuyucu Için OBD Araç Teşhis Aracı https://a.aliexpress.com/_EzASCjj

**Comment by Furtif** [comment]
for edit data fields you need unlock stuff 
![image](https://github.com/user-attachments/assets/0ca1171c-559a-47d8-aa73-39eb2da35346)

**Comment by enesokyay** [comment]
> Veri alanlarını düzenlemek için bir şeylerin kilidini açmanız gerekir ![görüntü](https://private-user-images.githubusercontent.com/11375035/360449883-0ca1171c-559a-47d8-aa73-39eb2da35346.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMzQwMDMsIm5iZiI6MTcyNDMzMzcwMywicGF0aCI6Ii8xMTM3NTAzNS8zNjA0NDk4ODMtMGNhMTE3MWMtNTU5YS00N2Q4LWFhNzMtMzllYjJkYTM1MzQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDEzMzUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjN2E3NzZhNGYwOGRlNThlNzJkYjk5M2E1MTBmMjJjMzc5Yzc5YmU3NmU4YmQ0M2U1MGE1MjQ5YzU3YjUwMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uWXUW0MuHKHVNEVaADCXueuq4mZw7A1Bz6-wksOtclg)

Don’t I need to first go to the editing screen to activate it? Because after scanning the ECUs with CAN, the entire system freezes. My computer takes almost a minute to respond to any clicks. From what I’ve seen in all the videos, it seems I need to press that button to make any corrections.

**Comment by Furtif** [comment]
> Je soupçonne que le problème pourrait être lié à l'adaptateur OBD, j'en ai donc commandé un nouveau sur AliExpress
> 
> 6,64€ | ELM327 V2.1 OBD2 Tarayıcı TR6260S1 BT/Wifi ELM 327 Android /IOS PK Vgate Icar2 Kod Okuyucu ici OBD Araç Teşhis Aracı https://a.aliexpress.com/_EzASCjj

c'est possible aussi, mais votre image de test ecus a l'air de bien fonctionner.

**Comment by Furtif** [comment]
> > Veri alanlarını düzenlemek için bir şeylerin kilidini açmanız gerekir![vue d'ensemble](https://private-user-images.githubusercontent.com/11375035/360449883-0ca1171c-559a-47d8-aa73-39eb2da35346.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMzQwMDMsIm5iZiI6MTcyNDMzMzcwMywicGF0aCI6Ii8xMTM3NTAzNS8zNjA0NDk4ODMtMGNhMTE3MWMtNTU5YS00N2Q4LWFhNzMtMzllYjJkYTM1MzQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDEzMzUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjN2E3NzZhNGYwOGRlNThlNzJkYjk5M2E1MTBmMjJjMzc5Yzc5YmU3NmU4YmQ0M2U1MGE1MjQ5YzU3YjUwMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uWXUW0MuHKHVNEVaADCXueuq4mZw7A1Bz6-wksOtclg)
> 
> Ne dois-je pas d'abord aller sur l'écran d'édition pour l'activer ? Car après avoir scanné les ECU avec CAN, tout le système se bloque. Mon ordinateur met presque une minute à répondre aux clics. D'après ce que j'ai vu dans toutes les vidéos, il semble que je doive appuyer sur ce bouton pour effectuer des corrections.

quand vous cliquez sur ```write``` sur n'importe quel setting a écrire il montre un message warning, pour debloquer faudra cliquer la.

**Comment by enesokyay** [comment]
> > Je soupçonne que le problème pourrait être lié à l'adaptateur OBD, j'en ai donc commandé un nouveau sur AliExpress
> > 6,64€ | ELM327 V2.1 OBD2 Tarayıcı TR6260S1 BT/Wifi ELM 327 Android /IOS PK Vgate Icar2 Kod Okuyucu ici OBD Araç Teşhis Aracı https://a.aliexpress.com/_EzASCjj
> 
> c'est possible aussi, mais votre image de test ecus a l'air de bien fonctionner.

Oui, c’est justement cela qui est étrange. Je pense qu’il y a un problème de vitesse avec l’appareil, c’est pourquoi j’en ai commandé un nouveau. Cependant, je n’ai aucun problème tant que je ne sélectionne pas un ECU dans la section du module de contrôle de la carrosserie. Dès que je sélectionne un ECU, l’ordinateur ralentit instantanément.

**Comment by enesokyay** [comment]
> > > Veri alanlarını düzenlemek için bir şeyin kilidini açmanız gerekir![vue d'ensemble](https://private-user-images.githubusercontent.com/11375035/360449883-0ca1171c-559a-47d8-aa73-39eb2da35346.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMzQwMDMsIm5iZiI6MTcyNDMzMzcwMywicGF0aCI6Ii8xMTM3NTAzNS8zNjA0NDk4ODMtMGNhMTE3MWMtNTU5YS00N2Q4LWFhNzMtMzllYjJkYTM1MzQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDEzMzUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjN2E3NzZhNGYwOGRlNThlNzJkYjk5M2E1MTBmMjJjMzc5Yzc5YmU3NmU4YmQ0M2U1MGE1MjQ5YzU3YjUwMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uWXUW0MuHKHVNEVaADCXueuq4mZw7A1Bz6-wksOtclg)
> > 
> >  Ne dois-je pas d'abord aller sur l'écran d'édition pour l'activer ? Car après avoir scanné les ECU avec CAN, tout le système se bloque. Mon ordinateur met presque une minute à répondre aux clics. D'après ce que j'ai vu dans toutes les vidéos, il semble que je doive appuyer sur ce bouton pour effectuer des corrections.
> 
> quand vous cliquez sur `write` sur n'importe quel setting a écrire il montre un message warning, pour debloquer faudra cliquer la.

C’est exactement ça le problème. Je ne peux pas accéder à l’écran de configuration pour écrire

**Comment by Furtif** [comment]
> > > Je soupçonne que le problème pourrait être lié à l'adaptateur OBD, j'en ai donc commandé un nouveau sur AliExpress 
> > > 6,64€ | ELM327 V2.1 OBD2 Tarayıcı TR6260S1 BT/Wifi ELM 327 Android /IOS PK Vgate Icar2 Kod Okyucu ici OBD Araç Teşhis Aracı https://a.aliexpress.com/_EzASCjj
> > 
> > 
> > c'est possible aussi, mais votre image de test ecus a l'air de bien fonctionner.
> 
> Oui, c'est justement cela qui est étrange. Je pense qu'il y a un problème de vitesse avec l'appareil, c'est pourquoi j'en ai commandé un nouveau. Cependant, je n'ai aucun problème tant que je ne sélectionne pas un ECU dans la section du module de contrôle de la carrosserie. Dès que je sélectionne un ECU, l'ordinateur ralentit instantanément.

double-click sur l'ecu que vous voulais puis section ecrans plus bas.

**Comment by Furtif** [comment]
car ici 
![image](https://github.com/user-attachments/assets/91f3d5b1-a087-4e5f-b163-6109008e6019)
 ne ne vois pas ligne dans section ecrans

**Comment by Furtif** [comment]
double-click la 
![image](https://github.com/user-attachments/assets/22917138-2a16-423f-a568-e514676fbbfd)
apres il aura dans la section ecrans les stuff pour edit.

**Comment by Furtif** [comment]
comme sa double-clik sur l'ecu 
![image](https://github.com/user-attachments/assets/04e7858f-0b58-4e70-b269-e4252083e35f)

**Comment by enesokyay** [comment]
Je n'arrive pas à expliquer le problème que j'ai. J'ai fait une vidéo mais je ne peux pas la télécharger ici.

**Comment by Furtif** [comment]
> Je n'arrive pas à expliquer le problème que j'ai. J'ai fait une vidéo mais je ne peux pas la télécharger ici.

oky

**Comment by enesokyay** [comment]
Lorsque j’arrive à l’écran qui indique ‘FENETRE ECRAN’, le programme ne fonctionne pas. Je ne peux effectuer aucune action. Cela prend environ une minute pour réagir.

**Comment by Furtif** [comment]
> Lorsque j’arrive à l’écran qui indique ‘FENETRE ECRAN’, le programme ne fonctionne pas. Je ne peux effectuer aucune action. Cela prend environ une minute pour réagir.

avais vous une BD bien ?

**Comment by enesokyay** [comment]
Je rentre chez moi et je vous ferai parvenir la vidéo.

**Comment by Furtif** [comment]
si c'est pas le cas https://github.com/cedricp/ddt4all/discussions/845 prennez celle-ci (ne dezipez pas placez le zip dans root de votre source python)

**Comment by Furtif** [comment]
sinon dans mon macOS tout fonctionne très bien, la seule différence c'est le CPU moi suis sur chipts Intel et vous dans M1 arm64, mais je ne crois pas que cela soit un problème.

**Comment by enesokyay** [comment]
https://github.com/user-attachments/assets/e5bebd05-efe0-43a2-91ab-51a65eaca335

**Comment by Furtif** [comment]
> trim.CDE46854-5ECE-4A7F-8F6F-43EB06E39E8B.MOV

vous parlais du délais a load list des écrans config ?, oui sa c'est normal un délai, il y a 4000 et quelque ecus e des milliers ecrans, mise a part une DB custom (selemente avec le vehicule) ou un ordinateur de la nasa peut etre plus rapide.

**Comment by enesokyay** [comment]
Le problème est que ces bégaiements ne disparaissent jamais. Je sais que j'ai attendu 30 minutes pour le premier scan de l'ecu. ensuite j'ai été patient et j'ai attendu à nouveau pour le faire de cette façon, mais je n'obtiens aucun résultat. si vous remarquez dans la vidéo, il n'y a aucun changement même si je bouge la souris. j'ai regardé des vidéos sur youtub et les gens attendent juste le premier démarrage.

**Comment by Furtif** [comment]
> Le problème est que ces bégaiements ne disparaissent jamais. Je sais que j'ai attendu 30 minutes pour le premier scan de l'ecu. ensuite j'ai été patient et j'ai attendu à nouveau pour le faire de cette façon, mais je n'obtiens aucun résultat. si vous remarquez dans la vidéo, il n'y a aucun changement même si je bouge la souris. j'ai regardé des vidéos sur youtub et les gens attendent juste le premier démarrage.

oui vu sa tester sur mon macOs j'ai pas autant délai .... chelou!

quand vous cliquez sur > pour deployer la section sa rame aussi ?

**Comment by enesokyay** [comment]
j'ai pu accéder à l'écran une fois (après environ 35 minutes). j'ai fait le même processus en regardant une vidéo sur youtub mais je n'ai pas pu obtenir de données parce que j'ai sélectionné le mauvais ecu dans mon véhicule. dois-je attendre 45 minutes pour chaque ecu afin de sélectionner le bon ecu ? ce processus ne devrait pas prendre autant de temps.

et Oui, il se bloque ainsi lorsque je sélectionne quoi que ce soit sur l'écran `ecu windows`.

**Comment by Furtif** [comment]
en mode Démo il y a le même problème? (j'ai remarquer que vous êtes en mode connecté, peut être sa prends du temps pendant que read les infos du ecu)

**Comment by Furtif** [comment]
Si en mode Démo rien de cella se passe, peut-être le problème est lie au périphérique OBD quand il tente de lire les informations du Ecu

**Comment by enesokyay** [comment]
Je n'ai aucun problème en mode démo.

Oui, j'ai pris une vidéo pendant que j'étais connecté à l'appareil. mais peu importe le temps que j'attends, ça ne s'améliore pas. je pensais que ça prenait du temps à charger. j'ai scanné l'ecu avec can pendant 35 minutes. puis j'ai attendu 30 minutes pour 1 écran d'ecu (malgré le scan) et j'ai vu que ce n'était pas le bon ecu. je n'ai même pas essayé d'essayer pour le deuxième, j'ai commandé un obd à nouveau et j'ai commencé un fil de discussion sur le forum.

**Comment by Furtif** [comment]
vais convertir ce issu en discussion,  si mode Démo aucun bug alors c'est le OBD que est mauvais.

**Comment by enesokyay** [comment]
J'y ai pensé aussi, mais à ce stade, c'est bizarre qu'il scanne tous les écus sur l'écran du terminal.

**Reply by Furtif** [reply]
oui le scan ne lit pas le contenu, il check seulement les adresse pas plus.

**Reply by Furtif** [reply]
regardez par la dans les discussions il y a quelques liens pour avoir un bon OBD

**Reply by enesokyay** [reply]
peut être à cause du processeur macos, peut être que j'ai fait une erreur dans le programme, peut être que l'obd est défectueux. Je ne sais pas, avez-vous regardé l'appareil obd que j'ai acheté sur aliexpress ? J'espère qu'il est de bonne qualité. Avez-vous vos informations dans un magasin (Carter-cash, etc.) ?

**Reply by Furtif** [reply]
sinon voir Readme il y a aussi les ODB compatibles testées ... j'ai un ODBLink EX il marche bien aussi mais je ne l'ai pas ajouter dans readme car a mon avis il y a que moi que l'a tester.

**Reply by Furtif** [reply]
> peut être à cause du processeur macos, peut être que j'ai fait une erreur dans le programme, peut être que l'obd est défectueux. Je ne sais pas, avez-vous regardé l'appareil obd que j'ai acheté sur aliexpress ? J'espère qu'il est de bonne qualité. Avez-vous vos informations dans un magasin (Carter-cash, etc.) ?

les miens je les ai acheter sur amazon. avec Pikxxxxxx comme mentionnée dans readme mise a part mon ODBLink EX que j'ai pris sa au pif, j'ai pas demontr pour voir le chip mais il fonctionne.

**Reply by enesokyay** [reply]
Ok, merci beaucoup pour votre aide, je vais essayer de changer quelques choses et j'écrirai ici.

**Reply by Furtif** [reply]
voir celui la [ODB all protocols](https://www.amazon.fr/Diagnostic-Voiture-Danalyse-PIC18F25K80-Protocoles/dp/B0C6H5HD2T/ref=sr_1_1_sspa?dib=eyJ2IjoiMSJ9.QxuazZCAtri4JIqhZC-ePPkRT1gu98Q2d0skkP4DS2IOuBF2yWxfG_9TIq5XN5hOqFGDTKgcOF45kMdQ6iLdOjJQmFL4KZA_0sojLAff_6lVPFUjNu7GNF9VEukhZ-bTyf5QpQRzua8eq3uLnPyv47ZmSA7E4lZQdStzjp-kZT5AmWxD4Tn7xkFaEHo3NKpncdPjho26IN-K6GRYgqbzjL5MHb7PaLXpETXAPc8rU66iBCznhVEc62YwX9maBN-DB43Y7yrEjn9xHVg-FIf9ud7gx_2yNZqBdXG4BZzZqJY.DkGxYlMm-Wk1GmpuLyP44NaFVJcJTcgbRoSk4T1VE4k&dib_tag=se&keywords=pic18f25k80&qid=1724337857&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1)

le mien est comme lui mais il y a pas pas le button levier  mais c'est un comme sa usb only, puis j'ai un autre wifi avec meme chip mais aussi sans le button levier. faudra demander si quelqu'un la tester.

**Reply by Furtif** [reply]
c'est dire que les OBD avec le button la levier je crois que font aussi les protocols CAN2 (multimedia) mais moi pour entrer dans multimedia can2 je utilise une rallonge comme dans ce discussion https://github.com/cedricp/ddt4all/discussions/970

**Reply by Furtif** [reply]
après pour arch MacOs (arm64, si python ok ) je ne crois pas que cella soit le problème, vu que mode Démo ok, si cella serait le problème DDT4ALL ouvrait même pas.

**Reply by Furtif** [reply]
Toute fois pouvez vou faire un test ELM avec le button?

![image](https://github.com/user-attachments/assets/79dbdd79-eac2-403b-913a-ab36b88fea99)

**Reply by Furtif** [reply]
voir si vous avez des results similaires:

```
v1.0 ATZ [OK]
v1.0 ATE1 [OK]
v1.0 ATL0 [OK]
v1.0 ATI [OK]
v1.0 AT@1 [OK]
v1.0 ATAL [OK]
v1.0 ATBD [OK]
v1.0 ATBI [OK]
v1.0 ATCAF0 [OK]
v1.0 ATCFC1 [OK]
v1.0 ATCP 01 [OK]
v1.0 ATCS [OK]
v1.0 ATCV 1250 [OK]
v1.0 ATD [OK]
v1.0 ATDP [OK]
v1.0 ATDPN [OK]
v1.0 ATH0 [OK]
v1.0 ATI [OK]
v1.0 ATIB 10 [OK]
v1.0 ATIB 96 [OK]
v1.0 ATL1 [OK]
v1.0 ATM0 [OK]
v1.0 ATMA [OK/TEMPS DEPASSÉ]
v1.0 ATMR 01 [OK/TEMPS DEPASSÉ]
v1.0 ATMT 01 [OK/TEMPS DEPASSÉ]
v1.0 ATNL [OK/TEMPS DEPASSÉ]
v1.0 ATPC [OK]
v1.0 ATR1 [OK]
v1.0 ATRV [OK]
v1.0 ATSH 001122 [OK]
v1.0 ATSH 012 [OK]
v1.0 ATSP A6 [OK]
v1.0 ATSP 6 [OK]
v1.0 ATCM 123 [OK]
v1.0 ATCM 12345678 [OK]
v1.0 ATCF 123 [OK]
v1.0 ATCF 12345678 [OK]
v1.0 ATST FF [OK]
v1.0 ATSW 96 [OK]
v1.0 ATSW 34 [OK]
v1.0 ATTP A6 [OK]
v1.0 ATTP 6 [OK]
v1.0 ATWM 817AF13E [OK]
v1.0 ATWS [OK]
v1.1 ATFC SD 300000 [OK]
v1.1 ATFC SH 012 [OK]
v1.1 ATFC SH 00112233 [OK]
v1.1 ATFC SM 1 [OK]
v1.1 ATPP FF OFF [OK]
v1.1 ATPP FF ON [OK]
v1.1 ATPPS [OK]
v1.2 ATAR [OK]
v1.2 ATAT1 [OK]
v1.2 ATDM1 [OK/TEMPS DEPASSÉ]
v1.2 ATIFR H [OK/TEMPS DEPASSÉ]
v1.2 ATIFR0 [OK]
v1.2 ATIIA 01 [OK]
v1.2 ATKW0 [OK]
v1.2 ATMP 0123 [OK/TEMPS DEPASSÉ]
v1.2 ATMP 0123 4 [OK/TEMPS DEPASSÉ]
v1.2 ATMP 012345 [OK/TEMPS DEPASSÉ]
v1.2 ATMP 012345 6 [OK/TEMPS DEPASSÉ]
v1.2 ATSR 01 [OK]
v1.3 ATCRA 012 [OK]
v1.3 ATCRA 01234567 [OK]
v1.3 ATD0 [OK]
v1.3 ATFE [OK]
v1.3 ATJE [OK]
v1.3 ATJS [OK]
v1.3 ATKW [OK]
v1.3 ATRA 01 [OK]
v1.3 ATS1 [OK]
v1.3 ATSP 00 [OK]
v1.3 ATV0 [OK]
v1.4 ATCEA [OK]
v1.4 ATCEA 01 [OK]
v1.4 ATCV 0000 [OK]
v1.4 ATIB 48 [OK]
v1.4 ATIGN [OK]
v1.4 ATPB 01 23 [OK]
v1.4 ATRD [OK]
v1.4 ATSD 01 [OK]
v1.4 ATSI [OK]
v1.4 ATFI [OK]
v1.4 ATSS [OK]
v1.4 ATTA 12 [OK]
v1.4 ATCSM1 [OK]
v1.4 ATJHF1 [OK]
v1.4 ATJTM1 [OK]
v1.4 ATJTM5 [OK]
v1.4b ATCRA [OK]
v2.0 ATAMC [ECHEC]
v2.0 ATAMT 20 [ECHEC]
v2.1 ATCTM1 [ECHEC]
v2.1 ATCTM5 [ECHEC]
Résultat: 91 réussis sur 95
Version ELM maximum:v1.4b
```

**Reply by Furtif** [reply]
le test en haut est fait avec ODBLink EX:

```
Tentative d'ouverture du port COM4 @ 115200
Serial<id=0x1f39bac3a30, open=True>(port='COM4', baudrate=115200, bytesize=8, parity='N', stopbits=1, timeout=5, xonxoff=False, rtscts=False, dsrdtr=False)
Connexion OBDLink OK, tentative de commutation UART à pleine vitesse
OBDLINK a commuté le baurate OK, changeant la vitesse UART maintenant...
Connexion pleine vitesse OBDLink OK
Version OBDLink STN2230 v5.10.3
```

je ne peut pas le refaire avec mon obd WIFI car il nécessite être connecter a la voiture pour que le wifi marche.

**Reply by enesokyay** [reply]
Je viens de commander ce produit, je pense que l'obd que j'ai est problématique, je l'essaierai quand je retournerai au véhicule.

**Reply by enesokyay** [reply]
https://www.amazon.fr/gp/product/B0CXW88FFM? ref=ppx_pt2_dt_b_prod_image

**Reply by Furtif** [reply]
> https://www.amazon.fr/gp/product/B0CXW88FFM? ref=ppx_pt2_dt_b_prod_image

soit sur qu'il a un chip PIC18F25K80 comme décrit dans readme !!!!

**Reply by enesokyay** [reply]
![image](https://github.com/user-attachments/assets/e329b8e0-1f9a-45d1-a215-fb89b1ea69bc) comme ça ?

**Reply by Furtif** [reply]
> ![image](https://private-user-images.githubusercontent.com/178347757/360549916-e329b8e0-1f9a-45d1-a215-fb89b1ea69bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNDQ0NDcsIm5iZiI6MTcyNDM0NDE0NywicGF0aCI6Ii8xNzgzNDc3NTcvMzYwNTQ5OTE2LWUzMjliOGUwLTFmOWEtNDVkMS1hMjE1LWZiODliMWVhNjliYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMlQxNjI5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYTQwMWNjNTBkMjY1ODE1MTJmZTg3ODRkYjU0NmUyZTVkMWI2NTkyN2UzOGUzODE1NzU0MjQ5Njg3MWFmNzNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MAeA_amkxJ4GyrFV3nK8YlxVuTKA55NuaSHWC_Qq6kw) comme ça ?

oui sa a tout l'air correct lui.. mai il y a d'autres comme celui que j'ai mis plus haut que fais dual avec un button levier

**Reply by Furtif** [reply]
Ou lui https://github.com/cedricp/ddt4all/issues/1201#issuecomment-2305087288 mais bon ... moi j'ai un simples comme plus ou moins celui de ton image après je fais le dual avec une rallonge câble croisé ... (mais sa c'est juste pour certains modèles Renault/dacia/nissan)

**Reply by enesokyay** [reply]
Je pense que cela sera utile pour fermer les fenêtres avec la télécommande de ma Clio 4. Je l'aurai demain, je vais l'essayer et le partager.

**Reply by AndyIreland** [reply]
> I think this will be useful to close the windows with the remote control of my Clio 4. I will have it tomorrow, I will try it and share it.

dont think you can without a factory installed Renault car alarm or some extra device to close the windows

**Reply by enesokyay** [reply]
> > I think this will be useful to close the windows with the remote control of my Clio 4. I will have it tomorrow, I will try it and share it.
> 
> 
> 
> dont think you can without a factory installed Renault car alarm or some extra device to close the windows

I saw that there are options that are not active in many vehicles, even if they are out of the factory on the internet, in many YouTube videos.

**Reply by AndyIreland** [reply]
indeed there are ... but even when you enable some that are disabled it does not mean they are going to work still. auto window closing being one such feature if you enable it , it still doesnt work I think you will find

![image](https://github.com/user-attachments/assets/be340093-0bd6-45ac-affd-c92e3af0ecf4)

**Comment by enesokyay** [comment]
The problem is caused by the elm device. I'm having a connection problem because I got wifi alone this time.
